src/librustc_expand/base.rs

```diff
@@ -939,6 +939,7 @@ pub struct ExpansionData {
 /// when a macro expansion occurs, the resulting nodes have the `backtrace()
 /// -> expn_data` of their expansion context stored into their span.
 pub struct ExtCtxt<'a> {
+    pub session: &'a Session,
```

Comment:
Could you avoid introducing the dependency of `ExtCtxt` on `Session`?
The `span_debug: bool` field can be put into `ExpansionConfig`, for example.
It only needs to be actively set in `librustc_interface/passes.rs`, other `ExpansionConfig` don't have anything to do with proc macro server.